--- origsrc/vinagre-3.2.2/configure.ac	2011-11-14 11:03:39.000000000 -0600
+++ src/vinagre-3.2.2/configure.ac	2011-11-16 18:57:52.017744700 -0600
@@ -25,7 +25,7 @@ AC_CANONICAL_HOST
 
 AC_MSG_CHECKING([whether to build for a Windows host])
 AS_CASE([$host_os],
-  [*mingw*|*cygwin*], [gio_os=gio-windows-2.0; windows_host=yes],
+  [*mingw*], [gio_os=gio-windows-2.0; windows_host=yes],
   [gio_os=gio-unix-2.0; windows_host=no])
 AC_MSG_RESULT([$windows_host])
 
@@ -128,7 +128,7 @@ AC_TYPE_UINT64_T
 # Checks for library functions. See vinagre/pty_open.c for usage.
 AC_CHECK_FUNCS([getpt posix_openpt grantpt unlockpt ptsname ptsname_r])
 
-AM_CONDITIONAL([VINAGRE_PTY_OPEN], [test "x$ac_cv_func_getpt" = xyes])
+AM_CONDITIONAL([VINAGRE_PTY_OPEN], [test "x$ac_cv_func_getpt$ac_cv_func_posix_openpt" != xnono])
 
 AC_SEARCH_LIBS([strerror], [cposix])
 
